adam/language_specific/english/english_language_generator.py

```diff
@@ -142,10 +168,22 @@ def _translate_object_to_noun(
             if (dependency_node.part_of_speech != PROPER_NOUN) and (
                 MASS_NOUN not in lexicon_entry.properties
             ):
-                determiner_node = DependencyTreeToken("a", DETERMINER)
-                self.dependency_graph.add_edge(
-                    determiner_node, dependency_node, role=DETERMINER_ROLE
-                )
+                if count == 1:
+                    determiner_node = DependencyTreeToken("a", DETERMINER)
+                    self.dependency_graph.add_edge(
```

Comment:
Note this part is the same for all branches, so you can pull the same trick as above, calling it `determiner_node` or something.

Reply:
Done